### PR TITLE
feat: Add support for multiple tracer providers

### PR DESCRIPTION
**Pull request by mohit-a21**
Added InitService which internally creates a new TracerProvider
used for creating services with a different set of resource attributes
then the default service.

### CODE REVIEW

File: instrumentation/opentelemetry/init.go [hunk 1]
```diff
@@ -59,37 +77,80 @@ func Init(cfg *config.AgentConfig) func() {
 		log.Fatal(err)
 	}
 
+	defaultBatcher := sdktrace.NewBatchSpanProcessor(zipkinExporter)
```
Comment (jcchavezs) [inline instrumentation/opentelemetry/init.go:80]: I would not call it defaultBatcher as it is the only batcher. How about just calling it batcher?

File: instrumentation/opentelemetry/init.go [hunk 2]
```diff
@@ -59,37 +77,80 @@ func Init(cfg *config.AgentConfig) func() {
...
+	traceProviders = make(map[string]*sdktrace.TracerProvider)
+	batcher = defaultBatcher
+	sampler = defaultSampler
+	initialized = true
 	return func() {
+		mu.Lock()
+		defer mu.Unlock()
+		batcher.Shutdown(context.Background())
+		for _, tracerProvider := range traceProviders {
+			tracerProvider.Shutdown(context.Background())
```
Comment (jcchavezs) [inline instrumentation/opentelemetry/init.go:108]: I think you can also delete the traceprovider from the map.

Comment (mohit-a21) [reply instrumentation/opentelemetry/init.go:108]: Done.

File: instrumentation/opentelemetry/init.go (hunk 1 above)
Comment (mohit-a21) [reply instrumentation/opentelemetry/init.go]: Done. Changed the global batcher to `globalBatcher`.

**Review by jcchavezs — APPROVED** [review]

**Comment by jcchavezs** [comment]
Great work @mohit-a21!

**Pull request merged**